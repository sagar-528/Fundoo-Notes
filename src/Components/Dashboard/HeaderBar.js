import React, { Component } from 'react'
import  { View, TouchableOpacity } from 'react-native'
import { Appbar, FAB, Avatar } from 'react-native-paper';
import NoteCss from '../../Styles/NoteCss'

class HeaderBar extends Component {

    constructor(props) {
        super(props)
    
        this.state = {
            listView : true
        }
    }
    
    selectView = () => {
        const {onPress} = this.props
        this.setState({
          listView : !this.state.listView
       })
        // onPress()
   }

    render() {
        return (
            <View style = {NoteCss.container}>
                <Appbar.Header style={{ backgroundColor: 'grey'}}>
                    <Appbar.Action
                    icon = 'menu'
                    onPress = {() => this.props.navigation.openDrawer()}
                    />
                    <Appbar.Content
                    style = {NoteCss.appbar_content_style}
                    title = "Search your notes"/>
                    <Appbar.Action
                    icon = {this.state.listView ? 'view-grid-outline' : 'view-agenda-outline'}
                    onPress={this.selectView}
                    />
                    <TouchableOpacity
                    style = {NoteCss.avatar_style} >
                    <Avatar.Image size={25} source={require('../../Assets/profile.jpg')} />
                    </TouchableOpacity>
                </Appbar.Header>
            </View>
        )
    }
}


export default HeaderBar
