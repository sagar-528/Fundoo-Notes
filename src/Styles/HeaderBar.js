import { StyleSheet } from 'react-native'

const HeaderBarStyle = StyleSheet.create({
    container: {
        flex : 1
      },
  
    appbar_content_style : {
        fontSize: 18
    },
  
    avatar_style : {
        marginRight : 15, 
        marginLeft : 5
    },
})

export default HeaderBarStyle