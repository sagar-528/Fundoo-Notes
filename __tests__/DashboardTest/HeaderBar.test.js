import React from 'react';
import {shallow} from 'enzyme';
import HeaderBar from '../../src/Components/Dashboard/HeaderBar'

describe('test HeaderBar', () => {
    it('should match to snapshot', () => {
        const component = shallow(<HeaderBar/>)
        expect(component).toMatchSnapshot();
    })

    it('When onPress of Menu icon button it will open the Navigation Drawer', async () => {
        const navigation = { openDrawer : jest.fn() }
        const onPressEvent = jest.fn();
        const component = shallow(<HeaderBar onPress = {onPressEvent} navigation = {navigation}/>)
        const instance = component.instance();
        await instance.handleMenuButton();
        expect(onPressEvent).toHaveBeenCalled();
        expect(navigation.openDrawer).toHaveBeenCalled()
    })

})