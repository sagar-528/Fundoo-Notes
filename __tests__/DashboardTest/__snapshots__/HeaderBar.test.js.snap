// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test HeaderBar should match to snapshot 1`] = `ShallowWrapper {}`;

exports[`test NoteScreen should match to snapshot 1`] = `ShallowWrapper {}`;
